test/test_linearizer_failures.py

```diff
@@ -26,27 +26,40 @@ def test_failure_2(self):
     lin = Linearizer(ast)
     assert fuzz_linearizer(lin) != "PASS"
 
-  @unittest.skipUnless(Device.DEFAULT in ["METAL", "GPU", "LLVM"], "fails on these backends")
+  @unittest.skipUnless(Device.DEFAULT in ["METAL", "GPU"], "fails on these backends")
   def test_failure_3(self):
     ast = LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=BufferOps.MEM, src=(), arg=MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(32, 8, 16, 16), strides=(2048, 256, 16, 1), offset=0, mask=None, contiguous=True),)))),), arg=(32, 8, 16, 1))
     lin = Linearizer(ast)
     # METAL: AssertionError: Error Domain=AGXMetalG13X Code=3 "Threadgroup memory size (65536) exceeds the maximum threadgroup memory allowed (32768)" UserInfo={NSLocalizedDescription=Threadgroup memory size (65536) exceeds the maximum threadgroup memory allowed (32768)}
     assert fuzz_linearizer(lin) != "PASS"
 
-  @unittest.skipUnless(Device.DEFAULT in ["METAL", "LLVM"], "fails on these backends")
+  @unittest.skipUnless(Device.DEFAULT in ["METAL"], "fails on these backends")
   def test_failure_4(self):
     ast = LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=BufferOps.MEM, src=(), arg=MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(1, 1, 1, 4, 1, 12, 2, 29), strides=(0, 0, 0, 2, 0, 216, 1, 8), offset=0, mask=((0, 1), (0, 1), (0, 1), (0, 4), (0, 1), (0, 11), (0, 2), (0, 27)), contiguous=False), View(shape=(1, 1, 1, 4, 22, 84), strides=(0, 0, 0, 696, 58, 1), offset=0, mask=((0, 1), (0, 1), (0, 1), (0, 4), (0, 12), (0, 58)), contiguous=False), View(shape=(1, 1, 1, 4, 2, 11, 3, 28), strides=(0, 0, 0, 1848, 924, 84, 28, 1), offset=0, mask=None, contiguous=True))))),), arg=(1, 1, 1, 4, 1, 11, 1, 28))
     lin = Linearizer(ast)
     # related to OptOps.NOLOCALS
     # IndexError: list index out of range
     assert fuzz_linearizer(lin) != "PASS"
 
-  @unittest.skipUnless(Device.DEFAULT in ["CLANG", "LLVM"], "fails on these backends")
+
+class TestLinearizerFailuresFixed(unittest.TestCase):
+  @unittest.skipUnless(Device.DEFAULT in ["LLVM"], "fixed on these backends")
+  def test_failure_3(self):
+    ast = LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=BufferOps.MEM, src=(), arg=MemBuffer(idx=1, dtype=dtypes.float, st=ShapeTracker(views=(View(shape=(32, 8, 16, 16), strides=(2048, 256, 16, 1), offset=0, mask=None, contiguous=True),)))),), arg=(32, 8, 16, 1))
```

Comment:
cool i like the regression test. can you write it in a way that reuses the ast and not copied?

Reply:
better?